develop/interoperability/best-practices-for-teleporting-assets.md

```diff
@@ -0,0 +1,190 @@
+
+
+- [XCM Payment API](https://paritytech.github.io/polkadot-sdk/master/xcm_runtime_apis/fees/trait.XcmPaymentApi.html){target=\_blank} for cross-chain delivery fees:
+
+    ```typescript
+    --8<-- 'code/develop/interoperability/best-practices-for-teleporting-assets/xcm-delivery-fees.ts'
+    ```
+  This code calculates the delivery fees required to send an XCM message to a specific destination parachain, helping estimate cross-chain transaction costs.
+
+
+- [Asset Conversion API](https://paritytech.github.io/polkadot-sdk/master/pallet_asset_conversion/trait.AssetConversionApi.html){target=\_blank} (available on Asset Hub) for fee conversion:
+
+    ```typescript
+    --8<-- 'code/develop/interoperability/best-practices-for-teleporting-assets/asset-conversion-api.ts'
+    ```
+  This code connects to Asset Hub's `AssetConversionApi`. It calculates the exchange rate between different assets, enabling users to determine the amount of one token required to pay fees in another token.
+
+### Multi-Asset Fee Handling
+
+For dealing with non-sufficient assets and fees, there are different fee payment strategies you can implement:
+
+1. **[Asset Conversion](/polkadot-protocol/architecture/system-chains/asset-hub/#non-sufficient-assets){target=\_blank}**: Convert part of the transfer amount to pay fees.
+2. **Separate Fee Payment**: Use a different sufficient asset for fees.
+3. **Fee Sponsorship**: Have another account pay fees on behalf of the user.
+
+## Asset Type Considerations
+
+### Sufficient Assets
+
+Sufficient assets can:
+
+- Suffice for account existence by meeting the existential deposit requirements.
+- Pay for transaction fee, and, if the chain configuration allows, sufficient assets can also be used for XCM execution and delivery fees.
+
+!!!note Sufficient Assets
+    Always check the chain to determine what assets are considered sufficient.
+    This may vary from chain to chain.
+
+### Non-Sufficient Assets
+
+Non-sufficient assets can:
+
+- Pay for transaction and XCM execution/delivery fees by swapping for a sufficient asset.
+- Be used to create new accounts by swapping for an exact amount of a sufficient asset to cover the existential deposit of the new account.
+
+!!!note Swapping Non-Sufficient Assets
+    Swapping non-sufficient assets for a sufficient asset can be done with the help of [asset conversion](https://wiki.polkadot.network/learn/learn-asset-conversion-assethub){target=\_blank}, which is live on Asset Hub. Always ensure that there is an associated liquidity pool with healthy liquidity for the pair that you intend to swap.
+
+Always verify that the amount expected to be received in the recipient account is the amount expected after all fees are deducted (transaction fees, XCM fees, and/or swapping fees).
+
+From a UX perspective, when designing cross-chain applications, consider who will be paying for the fees.
+
+**Example scenario**: Sending 100 USDT to a new Asset Hub account.
```

Comment:
I didn't realize these were the same. Could be worth doing a text snippet. Don't want to keep adding to this PR though, but something to consider

Reply:
I removed the one under polkadot-protocol/architecture/system-chains/asset-hub.md as the design suggestion is not specific to Asset Hub. It is better to have it one place under best practices.